### PR TITLE
resolves 35 which to automatically set stsaus at due times

### PR DESCRIPTION
**Pull request by AnointingMax**

### CODE REVIEW

**Review by danieliheonu — APPROVED** [review]
Nice work bro

File: app/routes.py [hunk 1]
```diff
@@ -376,3 +376,32 @@ def remove_candidate():
 @app.route("/404", methods=["GET"])
 def missing_route():
     return render_template("404.html")
+
+
+# the update status function is the function to be ran repaeatedly hence the id
+# the function will be ran in 30 seconds intervals
+@scheduler.task('interval', id='do_update_status', seconds=10, misfire_grace_time=900)
```
Comment (LeanKhan) [inline app/routes.py:383]: Nice work. But we don't need this function to keep on running every 30 secs for all elections. There could be huge performance drawbacks. Why can't we set a schedule event to fire on the time the owner set. As per can't the scheduler ring at a set time? If so then can't we set it to execute at the ending_at?

File: app/routes.py [hunk 2]
```diff
@@ -376,3 +376,32 @@ def remove_candidate():
...
+                # check if the status of the election is started
+                elif election.status == statuses[1]:
+                    #check if the current time is greater than the ending time of the election if yes set the status of the election to ended
+                    if election.ending_at < now:
+                        election.status_id = 3
+            except Exception as e:
+                print(str(e))
+            try:
+                db.session.commit()
+            except Exception as e:
```
Comment (LeanKhan) [inline app/routes.py:406]: Nice work with the try...except and roll backs.

File: app/routes.py [hunk 3]
```diff
@@ -107,8 +107,8 @@ def election(link):
         return redirect(url_for('missing_route'))
     if election.owner.id == current_user.id:
         statuses = Status.query.all()
-    return render_template('election.html', title=election.name, election=election, statuses=statuses)
-
+        return render_template('election.html', title=election.name, election=election, statuses=statuses)
+    return render_template('election.html', title=election.name, election=election)
```
Comment (LeanKhan) [inline app/routes.py:111]: No need for the second render_template I think

**Comment by AnointingMax** [comment]
There is a need for the second one
The first one only happens when the it's the owner accessing the route so
it has statuses while the other is is for when other people access the
route hence the lack of statuses

**Pull request merged**